core-chart/templates/argocd/operator.yaml

```diff
@@ -0,0 +1,163 @@
+      if [ -z "$ARGOCD_POD" ]; then
+        >&2 timestamped_echo "==> ERROR: Argo CD pod not found! ...retrying in $WAIT_BEFORE_RETRY seconds..."
+        sleep $WAIT_BEFORE_RETRY
+      fi
+    done
+    timestamped_echo "==> Argo CD pod found: $ARGOCD_POD"
+
+    timestamped_echo "Waiting for Argo CD pod to be ready..."
+
+    until kubectl -n $NS wait --for=condition=Ready pod/$ARGOCD_POD --timeout=${WAIT_BEFORE_RETRY}s
+    do
+      >&2 timestamped_echo "==> ERROR: Argo CD pod is not ready yet! ...retrying..."
+    done
+    timestamped_echo "==> Argo CD shows condition Ready."
+
+    timestamped_echo "Getting Argo CD admin password..."
+    ARGOCD_PASSWD=""
+    while [ -z "$ARGOCD_PASSWD" ]; do
+      ARGOCD_PASSWD=$(kubectl -n $NS get secret argocd-initial-admin-secret -o jsonpath="{.data.password}" 2> /dev/null || true)
+      if [ -z "$ARGOCD_PASSWD" ]; then
+        >&2 timestamped_echo "==> ERROR: Argo CD password secret not found! ...retrying in $WAIT_BEFORE_RETRY seconds..."
+        sleep $WAIT_BEFORE_RETRY
+      fi
+    done
+    timestamped_echo "==> Argo CD admin password retrieved."
+
+    timestamped_echo "Logging into Argo CD cli..."
+    while ! kubectl -n $NS exec $ARGOCD_POD -- argocd login "{{ .Release.Name }}-argocd-server.$NS" --username admin --password $(echo $ARGOCD_PASSWD | base64 -d) --insecure ; do
+      >&2 timestamped_echo "==> ERROR: Argo CD login failed! ...retrying in $WAIT_BEFORE_RETRY seconds..."
+      sleep $WAIT_BEFORE_RETRY
+    done
+    timestamped_echo "==> Argo CD login succesful."
+
+    timestamped_echo "Adding WDS clusters..."
+    {{- range $cp := .Values.WDSes }}
+
+      cp_name={{ $cp.name }}
+
+      timestamped_echo "==> Looking for Control Plane \"$cp_name\"..."
+      # Wait for the CP to exists and be ready
+      while [[ $(kubectl get controlplane $cp_name -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do
+        >&2 timestamped_echo "==> ERROR: Control Plane \"$cp_name\" is not ready! ...retrying in $WAIT_BEFORE_RETRY seconds..."
+        sleep $WAIT_BEFORE_RETRY
+      done
+      timestamped_echo "==> Control Plane \"$cp_name\" is ready."
+
+      # determine the secret name and namespace
+      key=$(kubectl get controlplane $cp_name -o=jsonpath='{.status.secretRef.inClusterKey}')
+      secret_name=$(kubectl get controlplane $cp_name -o=jsonpath='{.status.secretRef.name}')
+      secret_namespace=$(kubectl get controlplane $cp_name -o=jsonpath='{.status.secretRef.namespace}')
+
+      kubeconfig="/tmp/${cp_name}.kubeconfig"
+
+      timestamped_echo "==> Getting \"$key\" from \"$secret_name\" secret in \"$secret_namespace\" for control plane \"$cp_name\"... \"$kubeconfig\""
+      kubectl get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}" | base64 -d > "$kubeconfig"
+
+      timestamped_echo "==> Copying kubeconfig \"${cp_name}.kubeconfig\" to Argo CD pod \"$ARGOCD_POD\"..."

```

Comment:
Why use the duplicative and directory-less expression `\"${cp_name}.kubeconfig\"` here instead of `"$kubeconfig"` ?